### PR TITLE
update data source tests that compare lists of fields to compare all fields

### PR DESCRIPTION
**Pull request by danawillow**
This fixes the TestAccContainerClusterDatasource tests and future-proofs this one and the others. I didn't bother touching the tests that weren't already using this check format.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by danawillow** [comment]
Re-requesting review since I think this has changed enough since the last review.

File: third_party/terraform/utils/test_utils.go [hunk 1]
```diff
@@ -66,3 +69,40 @@ func toBool(attribute string) (bool, error) {
...
+
+func checkDataSourceStateMatchesResourceState(dataSourceName, resourceName string) func(*terraform.State) error {
+	return checkDataSourceStateMatchesResourceStateWithIgnores(dataSourceName, resourceName, map[string]struct{}{})
+}
+
+func checkDataSourceStateMatchesResourceStateWithIgnores(dataSourceName, resourceName string, ignoreFields map[string]struct{}) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		ds, ok := s.RootModule().Resources[dataSourceName]
+		if !ok {
+			return fmt.Errorf("root module has no resource called %s", dataSourceName)
```
Comment (rileykarson) [inline third_party/terraform/utils/test_utils.go]:
```suggestion
			return fmt.Errorf("can't find %s in state", dataSourceName)
```

File: third_party/terraform/utils/test_utils.go [hunk 2]
```diff
@@ -66,3 +69,40 @@ func toBool(attribute string) (bool, error) {
...
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("can't find %s in state", resourceName)
+		}
+
+		dsAttr := ds.Primary.Attributes
+		rsAttr := rs.Primary.Attributes
+
+		errMsg := ""
+		for k, attr := range rsAttr {
```
Comment (rileykarson) [inline third_party/terraform/utils/test_utils.go]: Can you comment with why it's safe to iterate only over the resource fields?

File: third_party/terraform/utils/test_utils.go [hunk 3]
```diff
@@ -66,3 +69,40 @@ func toBool(attribute string) (bool, error) {
...
+			if _, ok := ignoreFields[k]; ok {
+				continue
+			}
+			if dsAttr[k] != attr {
```
Comment (rileykarson) [inline third_party/terraform/utils/test_utils.go]: super nitpicky, but aesthetically I'm slightly in preference of `dsAttr[k] != rsAttr[k]` and similar.

File: third_party/terraform/utils/test_utils.go (hunk 1 above)
Comment (rileykarson) [reply third_party/terraform/utils/test_utils.go]: (especially because it's the datasource we can't find)

Comment (danawillow) [reply third_party/terraform/utils/test_utils.go]: Done

File: third_party/terraform/utils/test_utils.go (hunk 2 above)
Comment (danawillow) [reply third_party/terraform/utils/test_utils.go]: Done

File: third_party/terraform/utils/test_utils.go (hunk 3 above)
Comment (danawillow) [reply third_party/terraform/utils/test_utils.go]: Done

**Pull request merged**